module/VuFind/tests/integration-tests/src/VuFindTest/Mink/OaiTest.php

```diff
@@ -0,0 +1,170 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+declare(strict_types=1);
+
+namespace VuFindTest\Mink;
+
+use function count;
+
+/**
+ * OAI-PMH test class.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class OaiTest extends \VuFindTest\Integration\MinkTestCase
+{
+    /**
+     * Default OAI config settings
+     *
+     * @var array
+     */
+    protected $defaultOaiConfig = [
+        'OAI' => [
+            'identifier' => 'vufind.org',
+            'repository_name' => 'test repo',
+            'page_size' => 10,
+        ],
+    ];
+
+    /**
+     * Data provider describing OAI servers.
+     *
+     * @return array[]
+     */
+    public static function serverProvider(): array
+    {
+        return [
+            'auth' => ['/OAI/AuthServer'],
+            'biblio' => ['/OAI/Server'],
+        ];
+    }
+
+    /**
+     * Test that OAI-PMH is disabled by default.
+     *
+     * @param string $path URL path to OAI-PMH server.
+     *
+     * @return void
+     *
+     * @dataProvider serverProvider
+     */
+    public function testDisabledByDefault(string $path): void
+    {
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . $path);
+        $page = $session->getPage();
+        $this->assertEquals(
+            'OAI Server Not Configured.',
+            $page->getText()
+        );
+    }
+
+    /**
+     * Test that a verb is required when enabled.
+     *
+     * @param string $path URL path to OAI-PMH server.
+     *
+     * @return void
+     *
+     * @dataProvider serverProvider
+     */
+    public function testVerbRequired(string $path): void
+    {
+        $this->changeConfigs(['config' => $this->defaultOaiConfig]);
+        $rawXml = file_get_contents($this->getVuFindUrl() . $path);
+        $xml = simplexml_load_string($rawXml);
+        $this->assertEquals('Missing Verb Argument', $xml->error);
+    }
+
+    /**
+     * Test that an identify response is provided and includes an appropriate repository name.
+     *
+     * @param string $path URL path to OAI-PMH server.
+     *
+     * @return void
+     *
+     * @dataProvider serverProvider
+     */
+    public function testIdentifyResponseRepositoryName(string $path): void
+    {
+        $this->changeConfigs(['config' => $this->defaultOaiConfig]);
+        $rawXml = file_get_contents($this->getVuFindUrl() . $path . '?verb=Identify');
```

Comment:
I'm using file_get_contents to fetch XML since I couldn't quickly figure out a way to cleanly do this with Mink. I think the disadvantage to this approach is that we won't benefit from code coverage measurement on these requests. @EreMaijala, do you think we should just use Mink to visit the same URLs to push in coverage, or do you know of a better way to get XML directly from Mink? I tried [the approach described at this page](https://stackoverflow.com/questions/16667345/behat-mink-how-can-i-evaluate-a-xml-response) but it didn't work for me.

Reply:
In [OAuth2Test](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/OAuth2Test.php#L218) I simply used `\VuFindHttp\HttpService` to fetch JSON. I think it would work here just as well. :)

Reply:
Thanks, @EreMaijala! I put together a coverage-friendly HTTP GET method here: 5e332bc63c286cde804847b46837343320158c23 -- what do you think? Would we benefit from using this in OAuth2Test as well (and perhaps creating a POST version as well)? If so, I can open a follow-up PR.

Reply:
Nice! Yes, would be great to have a POST version for OAuth2Test. 👍

Reply:
I did some refactoring to traits in c48cfedd341c9720cdfda78ec7f8cae99fb517b2 to reduce redundancy and separate the HTTP logic from the Mink logic. After this is merged, I can open a follow-up for OAuth2Test refactoring.

Reply:
Looks good, though tests are still failing. :)

Reply:
Yep, just a php-cs-fixer issue. They should pass now. I wasn't worrying too much about that since I also still have to address @aleksip's feedback, and I'm sure I'll introduce more style errors before I fix them all. ;-)